packages/protocol/test/ERC1724/ZkAsset.js

```diff
@@ -126,394 +157,383 @@ contract.skip('ZkAsset', (accounts) => {
-
-            const firstTransferAmount = 30;
-            const depositProof = proof.joinSplit.encodeJoinSplitTransaction({
-                inputNotes: [],
-                outputNotes: notes.slice(0, 2),
-                senderAddress: accounts[0],
-                inputNoteOwners: [],
-                publicOwner: accounts[0],
-                kPublic: firstTransferAmount * -1,
-                validatorAddress: zkAsset.address,
-            });
-
-            const depositProofOutput = outputCoder.getProofOutput(depositProof.expectedOutput, 0);
-            const depositProofHash = outputCoder.hashProofOutput(depositProofOutput);
-
-            const fakeInputNoteOwners = aztecAccounts.slice(2, 4);
-
-            const secondTransferAmount = 0;
-            const noteTransfer = proof.joinSplit.encodeJoinSplitTransaction({
-                inputNotes: notes.slice(0, 2),
-                outputNotes: notes.slice(2, 4),
-                senderAddress: accounts[0],
-                inputNoteOwners: fakeInputNoteOwners,
-                publicOwner: accounts[1],
-                kPublic: secondTransferAmount,
-                validatorAddress: aztecJoinSplit.address,
-            });
-
-            const noteTransferProofOutput = outputCoder.getProofOutput(noteTransfer.expectedOutput, 0);
-            const noteTransferProofHash = outputCoder.hashProofOutput(noteTransferProofOutput);
-
-            await ace.publicApprove(zkAsset.address, depositProofHash, firstTransferAmount, { from: accounts[0] });
-            await ace.publicApprove(zkAsset.address, noteTransferProofHash, secondTransferAmount, { from: accounts[1] });
-
-            await zkAsset.confidentialTransfer(depositProof.proofData, depositProof.signatures);
-
-            await truffleAssert.reverts(zkAsset.confidentialTransfer(noteTransfer.proofData, noteTransfer.signatures));
+            const depositPublicValue = -30;
+            const withdrawalPublicValue = 0;
+
+            const {
+                depositInputNotes,
+                depositOutputNotes,
+                depositInputOwnerAccounts,
+                withdrawalInputNotes,
+                withdrawalOutputNotes,
+            } = await setupTwoProofTest(noteValues);
+
+            const depositProof = new JoinSplitProof(
+                depositInputNotes,
+                depositOutputNotes,
+                sender,
+                depositPublicValue,
+                publicOwner,
+            );
+
+            const depositData = depositProof.encodeABI(zkAsset.address);
+            const depositSignatures = depositProof.constructSignatures(joinSplitValidator.address, depositInputOwnerAccounts);
+
+            await ace.publicApprove(zkAsset.address, depositProof.hash, depositPublicValue, { from: accounts[0] });
+            await zkAsset.confidentialTransfer(depositData, depositSignatures);
+
+            const fakeInputNoteOwners = [secp256k1.generateAccount(), secp256k1.generateAccount()];
```

Comment:
In my refactor, the standard naming for this kind of test variables switched to `malformed` instead of `fake`.

Reply:
Updated the names